objectivec/src/ort_session.mm

```diff
@@ -13,12 +13,38 @@
 #import "src/ort_env_internal.h"
 #import "src/ort_value_internal.h"
 
+namespace {
+class AllocatorDeleter {
+ public:
```

Comment:
Confusing name as I expected it deleted the allocator. 

Not sure if it's possible to do something closer to what we have in allocator.h. 

```
template <typename T>
using IAllocatorUniquePtr = std::unique_ptr<T, std::function<void(T*)>>;
```

If not, this would more closely replicate the BufferDeleter setup (which is a bit ugly with a separate allocation followed by addition to a unique_ptr, and we really shouldn't need that and IAllocatorUniquePtr, and one day we'll clean that up :P) so could we name it BufferDeleter for consistency?

Reply:
It's only being used in one place now, so I'll just use it more directly. If it becomes useful in other places we can come up with a better name. It would be nice if the C++ API didn't require manually freeing the char*, but I didn't come up with a good way to change that.